tapgarden/custodian.go

```diff
@@ -86,6 +86,13 @@ type Custodian struct {
 	// events about new proofs being imported.
 	proofSubscription *fn.EventReceiver[proof.Blob]
 
+	// statusEventsSubs is a map of subscribers that want to be notified on
+	// new status events, keyed by their subscription ID.
+	statusEventsSubs map[uint64]*fn.EventReceiver[fn.Event]
```

Comment:
Non-blocking, but we could make this a `lntuils.SyncMap` which'll abstract away from some of the mutex usage.

Reply:
I'll look into implementing `SyncMap` in a separate PR.

Reply:
May allow us to reuse this subscriber setup + publish + etc. logic across multiple packages / state machines as well (if it becomes a bit more generic).